[gpt-oss] Fix unknown recipient none

baonudesifeizhai opened the pull request:
Fix #24170 
## Purpose
Add commentary channel handling to parse_remaining_state

- Fix missing commentary channel processing in parse_remaining_state function
- Handle None recipient case in streaming parser remaining state
- Ensure no content is lost when generation stops in commentary channel
- Complete the fix for issue #24170"
## Test Plan

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

baonudesifeizhai committed:
Fix 'Unknown recipient: None' error in harmony_utils.py

- Add handling for recipient=None case in commentary channel
- Treat None recipient as regular assistant message with reasoning content
- Resolves intermittent error when using Responses API with gpt-oss-20b
- Fixes issue #24170

baonudesifeizhai committed:
Add commentary channel handling to parse_remaining_state

- Fix missing commentary channel processing in parse_remaining_state function
- Handle None recipient case in streaming parser remaining state
- Ensure no content is lost when generation stops in commentary channel
- Complete the fix for issue #24170

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Optimize code based on review feedback

- Merge recipient=None case with python/browser cases to eliminate code duplication
- Simplify parse_remaining_state by removing redundant if/else branches
- Apply yapf formatting fixes
- Maintain same functionality while improving code quality

baonudesifeizhai committed:
Refine commentary channel handling in parse_remaining_state

- Add conservative content validation to avoid parser state issues
- Only process content that doesn't contain message markers
- Fix linting error: use 'not in' instead of 'not ... in'
- Apply yapf formatting fixes
- Maintain function structure while improving safety

baonudesifeizhai committed:
Merge branch 'vllm-project:main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Delete

baonudesifeizhai committed:
Add commentary channel handling in parse_remaining_state

- Handle commentary channel in parse_remaining_state to prevent content loss
- Add conservative content validation to avoid parser state corruption
- Only process content that doesn't contain message markers
- Apply yapf formatting fixes
- Fix 'unexpected tokens remaining in message header' errors

baonudesifeizhai committed:
Add error handling for parser.process() in StreamingHarmonyContext

- Catch exceptions from openai_harmony parser to prevent stream interruption
- Log parser errors but continue processing to maintain service availability
- Avoid updating parser state for failed tokens to prevent corruption
- Fix 'unexpected tokens remaining in message header' errors

baonudesifeizhai committed:
Fix Unknown recipient: None error with proper None handling

- Add unified handling for recipient=None at function start
- Handle recipient=None based on message channel:
  - commentary channel: treat as reasoning content
  - other channels: treat as regular message output
- Remove duplicate recipient=None handling in commentary branch
- Fix linting errors and apply yapf formatting
- Resolve original issue #24170

baonudesifeizhai committed:
Merge branch 'vllm-project:main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Fix commentary channel unknown recipient handling

- Replace ValueError for unknown recipients with reasoning content processing
- Ensures all unknown recipients in commentary channel are handled gracefully
- Fix linting errors and apply yapf formatting
- This should resolve remaining 'Unknown recipient' errors

baonudesifeizhai committed:
Add parser reset on error to prevent state corruption

- Reset StreamableParser when processing errors occur
- Prevents corrupted parser state from affecting subsequent requests
- Should resolve streaming failures and second request errors
- Add proper error logging for parser reset operations

heheda12345 reviewed (commented):
Thanks for the fix!

baonudesifeizhai committed:
Revert context.py changes - harmony parser doesn't raise errors in token processing

baonudesifeizhai committed:
Clean up context.py - remove parser error handling, use main branch version

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

JasonZhu1313 commented on vllm/entrypoints/harmony_utils.py:
```diff
@@ -179,7 +179,39 @@ def parse_output_message(message: Message) -> list[ResponseOutputItem]:
     output_items: list[ResponseOutputItem] = []
     recipient = message.recipient
-    if recipient is not None and recipient.startswith("browser."):
+    if recipient is None:
```
I have tested channel = analysis , recipient is None; And it worked very well on my internal benchmark, I got o4-mini comparable performance! I have another case where the channel is None lol. Will create an issue and a potential fix for it!

baonudesifeizhai committed:
Fix unknown recipient None error in harmony_utils

- Handle recipient=None cases in commentary channel as preambles
- Map analysis channel to reasoning regardless of recipient value
- Follow OpenAI Harmony specification for channel handling
- Fixes issue #24283 and related 400 Bad Request errors

This addresses the bug where messages in commentary channel with
recipient=None would cause 400 errors. According to OpenAI Harmony
docs, these should be treated as user-visible preambles.

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Fix recipient=None handling in commentary channel

- Handle recipient=None cases in commentary channel as reasoning
- Prevents 400 BadRequestError when models generate commentary without explicit recipients
- Addresses issue #24283: GPT-OSS robust handling of commentary channel messages
- Maintains compatibility with existing OpenAI Harmony message format

This fixes the error 'Unknown recipient: None' that occurred when
gpt-oss models generated reasoning content in commentary channel
without specifying a recipient.

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Merge branch 'vllm-project:main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Fix

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

heheda12345 reviewed (approved):
LGTM! Thanks for the fix.

yeqcharlotte reviewed (changes requested):
thanks for the change. could you add some unit tests that cover the logic?

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai commented:
<img width="1780" height="713" alt="image" src="https://github.com/user-attachments/assets/3599eb33-c94d-46c8-bce6-1993822dbdac" />
 curl.exe -H "@03-400-request-headers.txt" --data "@03-400-request.json" --compressed -X POST "https://aiq10a8m0umlqc-8000.proxy.runpod.net/v1/responses" --output response_readable.json -v  still work after changed

baonudesifeizhai commented on vllm/entrypoints/harmony_utils.py:
```diff
@@ -316,6 +317,22 @@ def parse_output_message(message: Message) -> list[ResponseOutputItem]:
                 output_items.append(response_item)
         elif recipient is not None and (recipient.startswith("python")
                                         or recipient.startswith("browser")):
```
if recipient is not None and recipient.startswith("functions."):   .... we deal with custom function before.....

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

baonudesifeizhai committed:
Merge branch 'main' into fix-unknown-recipient-none

Signed-off-by: Dongjie Zou <85092850+baonudesifeizhai@users.noreply.github.com>
